### DISCUSSION TITLE
#9718 - Cable path trace without PathEndpoint (Redux)

### DISCUSSION THREAD

**Discussion by DanSheps** [netbox-community/netbox#12094] [Ideas]
Re-investigating #9718 as I recently was able to monkey patch a few of the functions to properly support this but we should have a solid approach to this first.

There are a couple different ways we can do this:

1. Standalone plugin to obtain the functionality, however it would be a lot of the same code
2. Alter code to allow for dynamic paths to be generated on the fly.  There are a couple of sub options:
  a. From any port
  b. From an unconnected front port so there is "directionality" and position
  c. From an unconnected port, front or rear port and when working with a rear port with multiple positions assume position 1
3. Alter code to allow for paths to be generated and saved.  Again, there are options, like above

I personally prefer option #2 b) as it makes it easy to determine the proper path in most instances as there is a direction (front port, to rear port) and a position (front port position).  a) and c) make things extremely messy.

**Comment by DanSheps** [comment]
With my monkey patch, I want to say most of the code was reused from the work from @tnn2 at https://github.com/tnn2/netbox/commit/e4081c56e23df7de83761a4936748d0ca84921a1

What I needed to do:

* Use the dynamic_path function almost as is (adjusted for new code)
* Add an position argument to from_origin to allow for a position to be sent from outside the function
* Add conditional to get_extra_context to allow for dynamic traces if no pre-formed cable path exists
* In svg\cables.py I made a few changes
  * Add conditional for the positional stack
  * Add a conditional for the near end to be drawn properly when starting at a FP.

For this testing, I didn't include any of the template code changes.

**Reply by v0tti** [reply]
How do you achieve the cable trace from an unconnected front port without template code changes? Unconnected front ports have no clickable cable trace button or am I missing something?

**Reply by DanSheps** [reply]
I was manually visiting the trace page.

It would require template changes to get this into core.

**Reply by DanSheps** [reply]
I tihnk I might try and do this as a plugin first, and instead of overwriting anything, just have a separate link to it from each page.

**Reply by v0tti** [reply]
Any news on this discussion? From which page would the link to the cable trace with a plugin be accessible?

**Reply by DanSheps** [reply]
I haven't had time to revisit this yet.  I am still working on finding a way to properly get the links and place them into some form of object.

**Comment by v0tti** [comment]
I also just ran into this "issue" of not being able to do a cable trace without an endpoint. Judging by the amount of issues over the time it certainly seems to be in demand.

Maybe another and simpler approach could be to treat front ports marked as connected as PathEndpoints? This wouldn't solve the issue in all cases but I would assume it would be enough for most.
Our use case is to track cross connects in a DC where the end points (front ports) are in control of our customers. We would just mark those front ports as connected.

**Reply by johannwagner** [reply]
We also ran into the same issue while tracking cross connects in our DC.
